src/Passes/Source/ValidationPass/ValidationPassConfiguration.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+#include "Types/Types.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class ValidationPassConfiguration
+    {
+      public:
+        using Set = std::unordered_set<std::string>;
+        // Setup and construction
+        //
+
+        /// Setup function that attached the configuration to the ConfigurationManager.
+        void setup(ConfigurationManager& config)
+        {
+            config.setSectionName("Validation configuration", "");
+            config.addParameter(
+                allow_internal_calls_, "allow-internal-calls", "Whether or not internal calls are allowed.");
+        }
+
+        static ValidationPassConfiguration fromProfileName(String const& name)
+        {
+            auto ret = ValidationPassConfiguration();
+            if (name == "generic")
+            {
+                ret.allow_internal_calls_     = true;
+                ret.whitelist_external_calls_ = false;
+                ret.whitelist_opcodes_        = false;
+            }
+            else if (name == "base")
+            {
+                ret.allow_internal_calls_     = false;
+                ret.whitelist_external_calls_ = true;
+                ret.whitelist_opcodes_        = true;
+                ret.opcodes_                  = Set{"br", "call", "unreachable", "ret"};
+                ret.external_calls_           = Set{"__quantum__qis__mz__body",    "__quantum__qis__read_result__body",
```

Comment:
Suggest listing these in one column (rather than two).

Reply:
This is the will of `clang-format` - not sure what the correct parameters are to do that, but I am happy to update it if we can figure out what it is.